nixos/modules/virtualisation/containers.nix

```diff
@@ -368,6 +429,26 @@ in
               '';
             };
 
+            additionalCapabilities = mkOption {
+              type = types.listOf types.str;
+              default = [];
+              example = [ "CAP_NET_ADMIN" "CAP_MKNOD" ];
+              description = ''
+                Grant additional capabilities to the container.  See the
+                capabilities(7) and systemd-nspawn(1) man pages for more
+                information.
+              '';
+            };
+            enableTun = mkOption {
```

Comment:
It hides implementation details, people might don't know what is needed in order to use tun devices inside a container. As it only describe the semantics, in future more options might be required to enable tun devices container. In this respect it is comparable to systemd's hardening flags: PrivateDevices=true creates a private `/dev` with `/dev/null`, `/dev/zero` and `/dev/random`. However in future this list can be extended without breaking the semantic.